core/src/main/java/org/openedx/core/domain/model/Block.kt

```diff
@@ -117,11 +117,11 @@ data class EncodedVideos(
                 || fallback?.url != null
 
     val videoUrl: String
-        get() = mobileHigh?.url
-            ?: mobileLow?.url
-            ?: desktopMp4?.url
+        get() = fallback?.url
             ?: hls?.url
-            ?: fallback?.url
+            ?: desktopMp4?.url
+            ?: mobileHigh?.url
+            ?: mobileLow?.url
             ?: ""
```

Comment:
The `videoUrl` for streaming (not downloading) needs to be selected based on the `stream_priority` property of the `VideoInfo` model. Though we can tackle it in a different PR.

Reply:
the videoUrl field uses only in 2 places - in CourseUnitContainerAdapter (where I took the condition) and in CourseUnitContainerFragment where we have only one check - !encodedVideo.videoUrl.endsWith(“.m3u8”)

so this field is not used for offline purposes

Reply:
Yeap. Just to clarify, the `stream_priority` field is used to select a video URL from all the other URLs in the encoded video object based on the priority configured on the backend side.

Reply:
We can ignore this for now.

Reply:
got it